sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -119,14 +139,18 @@ public HttpHeaders put(String name, String value) {
      * @return The updated HttpHeaders object
      */
     public HttpHeaders set(String name, String value) {
+        return setInternal(formatKey(name), name, value);
```

Comment:
How often do we needlessly formatKey when the value is null? It's probably not worth paying for a null check here, or changing this to be Supplier-based, but it might be worth considering.

Reply:
This should happen very rarely, only reason null-safe names were kept around is that it would be a runtime breaking change.